### PR TITLE
Explain how to create and connect to a socket REPL in README.

### PR DESCRIPTION
**Pull request by austinhaas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,6 +103,23 @@ point. You can, however, change this behaviour by invoking such
...
 for the symbol you want to show the docstring for.
 
+#### Starting and connecting to a socket server
+
+For Leiningen, add the following option to your ~/.lein/profiles.clj or your project.clj:
+```
+:jvm-opts ["-Dclojure.server.repl={:port 5555 :accept clojure.core.server/repl}"]
+```
+
+Then run `lein repl` from within your project directory to start the REPL, and `C-c M-c RET localhost RET 5555` from within Emacs to connect.
```
Comment (arichiardi) [inline README.md]: Can we add:

> to connect or add the following to your `.dir-locals.el`:
>     ((nil . ((inf-clojure-lein-cmd . ("localhost" . 5555)))))

File: README.md [hunk 2]
```diff
@@ -103,6 +103,23 @@ point. You can, however, change this behaviour by invoking such
...
+
+For boot, export the environment variable BOOT_JVM_OPTIONS:
+```
+export BOOT_JVM_OPTIONS='-Dclojure.server.repl="{:port 5555 :accept clojure.core.server/repl}"'
```
Comment (arichiardi) [inline README.md]: Same as above, but with:

> ...add the following to your `.dir-locals.el`:
>     ((nil . ((inf-clojure-boot-cmd . ("localhost" . 5555)))))

**Comment by arichiardi** [comment]
Thanks for taking the time for this. It is indeed clearer like this instead of having the section in `Caveats`.

**Comment by arichiardi** [comment]
If you've run out of time don't worry, just let me know and I can finish this off :smile:

**Comment by arichiardi** [comment]
Thanks! I will add a commit to this one ok? Please don't rebase :smile:

**Comment by austinhaas** [comment]
Sounds good. Thanks for your help.

**Comment by arichiardi** [comment]
Right, cannot add, no worries, will merge, no thank you a lot for triggering this :smile:

**Pull request merged**